static/app/components/links/link.tsx

```diff
@@ -44,19 +63,23 @@ export interface LinkProps
  * A context-aware version of Link (from react-router) that falls
  * back to <a> if there is no router present
  */
-function BaseLink({disabled, to, ...props}: LinkProps): React.ReactElement {
+const Link = styled(({disabled, to, ...props}: LinkProps) => {
   const location = useLocation();
   to = normalizeUrl(to, location);
 
   if (!disabled && location) {
     return <RouterLink to={locationDescriptorToTo(to)} {...props} />;
   }
 
-  return <a href={typeof to === 'string' ? to : ''} {...props} />;
-}
+  return <Anchor href={typeof to === 'string' ? to : ''} {...props} />;
+})`
+  ${linkStyles}
+`;
 
-// Re-assign to Link to make auto-importing smarter
-const Link = styled(BaseLink)`
+export const Anchor = styled('a', {
```

Comment:
not sure if we should have two components for this. In TanStack router, you can just have `<Link to=“https://google.com” />` and it’ll just work. Of course `to` is type-safe and it knows what to do, so it doesn’t work with react router.

If we export two components, I would do `<InternalLink to=...>` and `<ExternalLink href=...>` to be explicit about it.

Reply:
I think that's what we should try and do here too. I suspect that as a consequence of this split decision, we have people who just use `<a href...>` for external links.